src/App.tsx

```diff
@@ -14,11 +14,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  startList: string[],
+  goods: string[],
+  showList: boolean,
+};
+class App extends React.Component<{}, State> {
+  state = {
+    startList: goodsFromServer,
+    goods: goodsFromServer,
+    showList: false,
+  };
+
+  openList = () => {
+    this.setState(state => ({
+      showList: !state.showList,
+    }));
+  };
+
+  isReverse = () => {
+    this.setState(state => {
+      const newList = [...state.goods];
+
+      return {
+        goods: newList.reverse(),
+      };
+    });
+  };
+
+  isSortedByAlph = () => {
+    this.setState(state => {
+      const newList = [...state.goods];
+
+      return {
+        goods: newList.sort(),
+      };
+    });
+  };
+
+  isReset = () => {
+    const { startList } = this.state;
+
+    this.setState({ goods: startList });
+  };
+
+  isSortedByLength = () => {
+    this.setState(state => {
+      const newList = [...state.goods];
+
+      return {
+        goods: newList.sort((fruit1, fruit2) => fruit1.length - fruit2.length),
+      };
+    });
+  };
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="app">
+        <div className="app__buttons">
+          {this.state.showList === false && (
+            <button
+              className="button is-primary is-light"
+              type="button"
+              onClick={this.openList}
+            >
+              Start
+            </button>
+          )}
+          <button
+            className="button is-warning is-light"
+            type="button"
+            onClick={this.isReverse}
+          >
+            Reverse
+          </button>
+          <button
```

Comment:
add one empty line between same level elements
```suggestion
          </button>
          
          <button
```